src/main/java/hudson/plugins/filesystem_scm/FolderDiff.java

```diff
@@ -115,57 +124,54 @@ public void setAllowDeleteList(Set<String> allowDeleteList) {
      * @param breakOnceFound
      *            to improve performance, we will return once we found the 1st new
      *            or modified file
-     * @param testRun
-     *            if true, will not sync file from source to destination, otherwise,
-     *            will sync files if new or modified files found
      * 
      * @return the list of new or modified files
      */
-    public List<Entry> getNewOrModifiedFiles(long time, boolean breakOnceFound, boolean testRun) {
+    public List<Entry> getNewOrModifiedFiles(long time, boolean breakOnceFound) {
         File src = new File(srcPath);
         File dst = new File(dstPath);
 
-        IOFileFilter dirFilter = ignoreHidden ? HiddenFileFilter.VISIBLE : TrueFileFilter.TRUE;
-        AndFileFilter fileFilter = createAntPatternFileFilter();
-        Iterator<File> it = (Iterator<File>) FileUtils.iterateFiles(src, fileFilter, dirFilter);
         ArrayList<Entry> list = new ArrayList<Entry>();
-        while (it.hasNext()) {
-            File file = it.next();
-            try {
-                String relativeName = getRelativeName(file.getAbsolutePath(), src.getAbsolutePath());
-                boolean newOrModified = false;
-                // need to change dst to see if there is such a file
-                File tmp = new File(dst, relativeName);
-                if (!tmp.exists()) {
-                    newOrModified = true;
-                    list.add(new Entry(relativeName, Entry.Type.NEW));
-                    log("New file: " + relativeName);
-                } else if (FileUtils.isFileNewer(file, time) || FileUtils.isFileNewer(file, tmp)) {
-                    newOrModified = true;
-                    list.add(new Entry(relativeName, Entry.Type.MODIFIED));
-                    log("Modified file: " + relativeName);
-                }
-                if (newOrModified) {
-                    if (breakOnceFound)
-                        return list;
-                    if (!testRun) {
-                        // FileUtils.copyFile(file, tmp);
+        if (src.isDirectory()) {

```

Comment:
Here I would just check whether it is a directory and fail the method otherwise. No strong opinion though